api/src/routes/certificate.test.ts

```diff
@@ -0,0 +1,373 @@
+        });
+        expect(response.status).toBe(500);
+      });
+
+      // Note: Email does not actually send (work) in development, but status should still be 200.
+      test('should send the certified email, if all current certifications are met', async () => {
+        await fastifyTestInstance.prisma.user.updateMany({
+          where: { email: defaultUserEmail },
+          data: {
+            completedChallenges: [
+              { id: 'bd7158d8c442eddfaeb5bd18', completedDate: 123456789 },
+              { id: '587d78af367417b2b2512b03', completedDate: 123456789 },
+              { id: '587d78af367417b2b2512b04', completedDate: 123456789 },
+              { id: '587d78b0367417b2b2512b05', completedDate: 123456789 },
+              { id: 'bd7158d8c242eddfaeb5bd13', completedDate: 123456789 }
+            ],
+            isRespWebDesignCert: false,
+            isJsAlgoDataStructCert: true,
+            isFrontEndLibsCert: true,
+            is2018DataVisCert: true,
+            isRelationalDatabaseCertV8: true,
+            isApisMicroservicesCert: true,
+            isQaCertV7: true,
+            isSciCompPyCertV7: true,
+            isDataAnalysisPyCertV7: true,
+            isInfosecCertV7: true,
+            isMachineLearningPyCertV7: true,
+            isCollegeAlgebraPyCertV8: true,
+            isFoundationalCSharpCertV8: true
+          }
+        });
+
+        const spy = jest.spyOn(fastifyTestInstance, 'sendEmail');
+
+        const response = await superRequest('/certificate/verify', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          certSlug: Certification.RespWebDesign
+        });
+
+        expect(spy).toHaveBeenCalled();
+        expect(response.status).toBe(200);
+      });
+
+      test('should return 200 if all went well', async () => {
+        await fastifyTestInstance.prisma.user.updateMany({
+          where: { email: defaultUserEmail },
+          data: {
+            completedChallenges: [
+              { id: 'bd7158d8c442eddfaeb5bd18', completedDate: 123456789 },
+              { id: '587d78af367417b2b2512b03', completedDate: 123456789 },
+              { id: '587d78af367417b2b2512b04', completedDate: 123456789 },
+              { id: '587d78b0367417b2b2512b05', completedDate: 123456789 },
+              { id: 'bd7158d8c242eddfaeb5bd13', completedDate: 123456789 }
+            ],
+            isRespWebDesignCert: false
+          }
+        });
+
+        const response = await superRequest('/certificate/verify', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          certSlug: Certification.RespWebDesign
+        });
+
+        const user = await fastifyTestInstance.prisma.user.findFirst({
+          where: { email: defaultUserEmail }
+        });
+
+        expect(user).toMatchObject({ isRespWebDesignCert: true });
+        expect(response.body).toMatchObject({
+          response: { message: 'flash.cert-claim-success' }
+        });
```

Comment:
This leaves a fair chunk of the body uncovered. Ideally we want to know what `completedChallenges` and `isCertMap` are.

Since the client mindlessly updates the user with all the properties, it would be good to know what they are.